src/profiling/meter_manager.cpp

```diff
@@ -101,39 +103,54 @@ nlohmann::json to_json(const measurement& mnt) {
     };
 }
 
-nlohmann::json to_json(const meter_manager& manager) {
+// Simple type for gathering distributed meter information
+struct meter_report {
+    std::vector<std::string> checkpoints;
+    unsigned num_domains;
+    std::string global_model;
```

Comment:
`global_model` is an odd name for this field: can we use something like `global_communicator` instead? In fact, this could just hold the enum value, with the conversion to string performed in `to_json()` — this allows the `meter_report` to be the purely programmatic representation, and the `to_json()` output to be the text serialized representation.